zio-http/src/main/scala/zio/http/codec/internal/EncoderDecoder.scala

```diff
@@ -98,87 +102,83 @@ private[codec] object EncoderDecoder                   {
     }
   }
 
-  private final case class Single[-AtomTypes, Value](httpCodec: HttpCodec[AtomTypes, Value])
-      extends EncoderDecoder[AtomTypes, Value] {
+  private final case class Single[-AtomTypes, Value](
+    httpCodec: HttpCodec[AtomTypes, Value],
+    outputTypes: Chunk[MediaType] = Chunk.empty,
+  ) extends EncoderDecoder[AtomTypes, Value] {
     private val constructor   = Mechanic.makeConstructor(httpCodec)
     private val deconstructor = Mechanic.makeDeconstructor(httpCodec)
 
     private val flattened: AtomizedCodecs = AtomizedCodecs.flatten(httpCodec)
 
-    private val jsonEncoders: Chunk[Any => Body]                          =
-      flattened.content.map { bodyCodec =>
-        val erased    = bodyCodec.erase
-        val jsonCodec = JsonCodec.schemaBasedBinaryCodec(erased.schema)
-        erased.encodeToBody(_, jsonCodec)
+    private val codecs: Map[String, MediaTypeCodec[_]] = {
```

Comment:
If I understand correctly this means the default would be that all endpoints support json, protobuf and a plain text encoding.
I'm not sure that's the best default - in many cases you only need to support one specific (let's say json) content type in which case it seems wasteful to generate the codecs for all the other supported content types. It can also be misleading to have text codec on by default while it only works for primitive types.

So:
- maybe the default should be json only?
- only those codecs should be derived which are actually used

Reply:
This was how I understood @jdegoes in #1506. That if you would have an unconfigured instance of `Endpoint`, we accept everything that we possibly can. And that we would make it configurable later on. But in the current version, it will already narrow down based on `withOutputTypes` and the configured `ContentType`.
For this case I guess, I could make the building of the codecs lazy and reverse the filter. So only build codecs for types that are relevant.